server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -0,0 +1,102 @@
+package cached_mysql
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/fleetdm/fleet/v4/server/datastore"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	redigo "github.com/gomodule/redigo/redis"
+	"github.com/pkg/errors"
+)
+
+type cachedMysql struct {
+	fleet.Datastore
+	datastore.Locker
```

Comment:
Could it make sense to move `datastore.Locker` into the `fleet.Datastore` interface? That could also remove some type assertions in the code above.

Reply:
It could, the separation was because we might implement this lock with Redis in the future, as this implementation is less good than redlock or other.

Reply:
But the way we are using it, it doesn't matter at this point. We'll need to refactor anyway... I'll merge them.